src/veritysetup/veritysetup.c

```diff
@@ -107,7 +109,12 @@ static int parse_options(const char *options) {
                 else if (streq(word, "panic-on-corruption"))
                         arg_activate_flags |= CRYPT_ACTIVATE_PANIC_ON_CORRUPTION;
 #endif
-                else if ((val = startswith(word, "root-hash-signature="))) {
+                else if ((val = startswith(word, "hash-offset="))) {
+
+                        r = safe_atou64(val, &arg_hash_offset);
+                        if (r < 0)
+                                return log_error_errno(r, "Failed to parse %s: %m", word);
```

Comment:
if this shall be multiples of sector size I think we can refuse early on here if the value isn't divisble by 512. While devices can use larger sector sizes, they'll always be multiple of this, and they cannot use smaller ones

Reply:
I will return error if `arg_hash_offset` % `0` is not zero.